src/lib/y2storage/proposal_settings.rb

```diff
@@ -86,5 +91,81 @@ def initialize
       @home_min_disk_size            = DiskSize.GiB(10)
       @home_max_disk_size            = DiskSize.unlimited
     end
+
+    # Overrides all the settings with values read from the YaST product features
+    # (i.e. values in /control.xml).
+    #
+    # Settings omitted in the product features are not modified. For backwards
+    # compatibility reasons, features with a value of zero are also ignored.
+    #
+    # Calling this method modifies the object
+    def read_product_features!
+      set_from_boolean_feature(:use_lvm, :proposal_lvm)
+      set_from_boolean_feature(:use_separate_home, :try_separate_home)
+      set_from_boolean_feature(:use_snapshots, :proposal_snapshots)
+      set_from_boolean_feature(:enlarge_swap_for_suspend, :swap_for_suspend)
+      set_from_boolean_feature(:root_subvolume_read_only, :root_subvolume_read_only)
+
+      set_from_size_feature(:root_base_disk_size, :root_base_size)
+      set_from_size_feature(:root_max_disk_size, :root_max_size)
+      set_from_size_feature(:home_max_disk_size, :vm_home_max_size)
+      set_from_size_feature(:min_size_to_use_separate_home, :limit_try_home)
+
+      set_from_integer_feature(:root_space_percent, :root_space_percent)
+      set_from_integer_feature(:btrfs_increase_percentage, :btrfs_increase_percentage)
```

Comment:
I double checked the old storage-ng and `btrfs_increase_percentage` is only discarded if `< 0`. Not if `<=0`. I will fix it.